In [1]:
from bs4 import BeautifulSoup

In [2]:
import requests

In [3]:
page=requests.get("https://www.imdb.com/india/top-rated-indian-movies")
page

<Response [200]>

In [4]:
page.content

b'\n\n\n<!DOCTYPE html>\n<html\n    xmlns:og="http://ogp.me/ns#"\n    xmlns:fb="http://www.facebook.com/2008/fbml">\n    <head>\n         \n\n        <meta charset="utf-8">\n        <meta http-equiv="X-UA-Compatible" content="IE=edge">\n\n    \n    \n    \n\n    \n    \n    \n\n            <style>\n                body#styleguide-v2 {\n                    background: no-repeat fixed center top #000;\n                }\n            </style>\n        <style>\n            body#styleguide-v2 #root {\n                box-shadow: none;\n            }\n        </style>\n\n\n\n        <script type="text/javascript">var IMDbTimer={starttime: new Date().getTime(),pt:\'java\'};</script>\n\n<script>\n    if (typeof uet == \'function\') {\n      uet("bb", "LoadTitle", {wb: 1});\n    }\n</script>\n  <script>(function(t){ (t.events = t.events || {})["csm_head_pre_title"] = new Date().getTime(); })(IMDbTimer);</script>\n        <title>Top Rated Indian Movies - IMDb</title>\n  <script>(function(t){ (t.

In [5]:
soup=BeautifulSoup(page.content)
soup

<!DOCTYPE html>
<html xmlns:fb="http://www.facebook.com/2008/fbml" xmlns:og="http://ogp.me/ns#">
<head>
<meta charset="utf-8"/>
<meta content="IE=edge" http-equiv="X-UA-Compatible"/>
<style>
                body#styleguide-v2 {
                    background: no-repeat fixed center top #000;
                }
            </style>
<style>
            body#styleguide-v2 #root {
                box-shadow: none;
            }
        </style>
<script type="text/javascript">var IMDbTimer={starttime: new Date().getTime(),pt:'java'};</script>
<script>
    if (typeof uet == 'function') {
      uet("bb", "LoadTitle", {wb: 1});
    }
</script>
<script>(function(t){ (t.events = t.events || {})["csm_head_pre_title"] = new Date().getTime(); })(IMDbTimer);</script>
<title>Top Rated Indian Movies - IMDb</title>
<script>(function(t){ (t.events = t.events || {})["csm_head_post_title"] = new Date().getTime(); })(IMDbTimer);</script>
<script>
    if (typeof uet == 'function') {
      uet("be", "LoadTitl

In [6]:
print(soup.prettify())

<!DOCTYPE html>
<html xmlns:fb="http://www.facebook.com/2008/fbml" xmlns:og="http://ogp.me/ns#">
 <head>
  <meta charset="utf-8"/>
  <meta content="IE=edge" http-equiv="X-UA-Compatible"/>
  <style>
   body#styleguide-v2 {
                    background: no-repeat fixed center top #000;
                }
  </style>
  <style>
   body#styleguide-v2 #root {
                box-shadow: none;
            }
  </style>
  <script type="text/javascript">
   var IMDbTimer={starttime: new Date().getTime(),pt:'java'};
  </script>
  <script>
   if (typeof uet == 'function') {
      uet("bb", "LoadTitle", {wb: 1});
    }
  </script>
  <script>
   (function(t){ (t.events = t.events || {})["csm_head_pre_title"] = new Date().getTime(); })(IMDbTimer);
  </script>
  <title>
   Top Rated Indian Movies - IMDb
  </title>
  <script>
   (function(t){ (t.events = t.events || {})["csm_head_post_title"] = new Date().getTime(); })(IMDbTimer);
  </script>
  <script>
   if (typeof uet == 'function') {
      uet("be"

In [7]:
names=soup.find_all('td',class_="titleColumn")
names

[<td class="titleColumn">
       1.
       <a href="/title/tt0093603/" title="Mani Ratnam (dir.), Kamal Haasan, Saranya Ponvannan">Nayakan</a>
 <span class="secondaryInfo">(1987)</span>
 </td>,
 <td class="titleColumn">
       2.
       <a href="/title/tt0367495/" title="Sundar C. (dir.), Kamal Haasan, Madhavan">Anbe Sivam</a>
 <span class="secondaryInfo">(2003)</span>
 </td>,
 <td class="titleColumn">
       3.
       <a href="/title/tt8176054/" title="Mari Selvaraj (dir.), Kathir, Anandhi">Pariyerum Perumal</a>
 <span class="secondaryInfo">(2018)</span>
 </td>,
 <td class="titleColumn">
       4.
       <a href="/title/tt0079221/" title="Hrishikesh Mukherjee (dir.), Amol Palekar, Bindiya Goswami">Golmaal</a>
 <span class="secondaryInfo">(1979)</span>
 </td>,
 <td class="titleColumn">
       5.
       <a href="/title/tt7391996/" title="Venkatesh Maha (dir.), Radha Bessy, Subba Rao Vepada">C/o Kancharapalem</a>
 <span class="secondaryInfo">(2018)</span>
 </td>,
 <td class="titleColumn"

In [8]:
Series_name=[]
for i in names:
    for j in i.find_all("a"):
        Series_name.append(j.text.replace("\n",""))
Series_name

['Nayakan',
 'Anbe Sivam',
 'Pariyerum Perumal',
 'Golmaal',
 'C/o Kancharapalem',
 'Apur Sansar',
 'Manichitrathazhu',
 'Kireedam',
 'Natsamrat',
 'Pather Panchali',
 '96',
 'Thevar Magan',
 'Black Friday',
 'Kumbalangi Nights',
 'Visaaranai',
 '3 Idiots',
 'Soorarai Pottru',
 'Jersey',
 'Taare Zameen Par',
 'Drishyam 2',
 'Asuran',
 'Thalapathi',
 'Kaithi',
 'Dangal',
 'Sarpatta Parambarai',
 'Devasuram',
 'Ratsasan',
 'Aparajito',
 'Jaane Bhi Do Yaaro',
 'Pyaasa',
 'Peranbu',
 'Vada Chennai',
 'Guide',
 'Thani Oruvan',
 'Kannathil Muthamittal',
 'Chupke Chupke',
 'Iruvar',
 'Spadikam',
 'Agent Sai Srinivasa Athreya',
 'Super Deluxe',
 'Mahanati',
 'Aruvi',
 'Drishyam',
 'Khosla Ka Ghosla!',
 'Shershaah',
 'Pudhu Pettai',
 'Vikram Vedha',
 'Tumbbad',
 'Premam',
 'Anniyan',
 'Bangalore Days',
 'Mudhalvan',
 'Kaakkaa Muttai',
 'Anand',
 'Satya',
 'Andhadhun',
 'Papanasam',
 'Dhuruvangal Pathinaaru',
 'Soodhu Kavvum',
 'Shahid',
 'Jigarthanda',
 'Pithamagan',
 'Gangs of Wasseypur',
 'Pa

In [9]:
ratings=soup.find_all('td',class_="ratingColumn imdbRating")
ratings

[<td class="ratingColumn imdbRating">
 <strong title="8.5 based on 18,179 user ratings">8.5</strong>
 </td>,
 <td class="ratingColumn imdbRating">
 <strong title="8.5 based on 18,753 user ratings">8.5</strong>
 </td>,
 <td class="ratingColumn imdbRating">
 <strong title="8.5 based on 12,110 user ratings">8.5</strong>
 </td>,
 <td class="ratingColumn imdbRating">
 <strong title="8.5 based on 18,665 user ratings">8.5</strong>
 </td>,
 <td class="ratingColumn imdbRating">
 <strong title="8.5 based on 5,830 user ratings">8.5</strong>
 </td>,
 <td class="ratingColumn imdbRating">
 <strong title="8.5 based on 13,391 user ratings">8.5</strong>
 </td>,
 <td class="ratingColumn imdbRating">
 <strong title="8.5 based on 9,689 user ratings">8.5</strong>
 </td>,
 <td class="ratingColumn imdbRating">
 <strong title="8.5 based on 6,935 user ratings">8.5</strong>
 </td>,
 <td class="ratingColumn imdbRating">
 <strong title="8.4 based on 6,087 user ratings">8.4</strong>
 </td>,
 <td class="ratingColum

In [10]:
Series_rating=[]
for i in ratings:
     for j in i.find_all("strong"):
        Series_rating.append(j.text.replace("\n",""))
    
Series_rating

['8.5',
 '8.5',
 '8.5',
 '8.5',
 '8.5',
 '8.5',
 '8.5',
 '8.5',
 '8.4',
 '8.4',
 '8.4',
 '8.4',
 '8.4',
 '8.4',
 '8.4',
 '8.4',
 '8.3',
 '8.3',
 '8.3',
 '8.3',
 '8.3',
 '8.3',
 '8.3',
 '8.3',
 '8.3',
 '8.3',
 '8.3',
 '8.3',
 '8.3',
 '8.3',
 '8.3',
 '8.3',
 '8.3',
 '8.3',
 '8.2',
 '8.2',
 '8.2',
 '8.2',
 '8.2',
 '8.2',
 '8.2',
 '8.2',
 '8.2',
 '8.2',
 '8.2',
 '8.2',
 '8.2',
 '8.2',
 '8.2',
 '8.2',
 '8.2',
 '8.2',
 '8.2',
 '8.2',
 '8.2',
 '8.2',
 '8.2',
 '8.2',
 '8.2',
 '8.2',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.1',
 '8.0',
 '8.0',
 '8.0',
 '8.0',
 '8.0',
 '8.0',
 '8.0',
 '8.0',
 '8.0',
 '8.0',
 '8.0',
 '8.0',
 '8.0',
 '8.0',
 '8.0',
 '8.0',
 '8.0',
 '8.0',


In [11]:
years=soup.find_all('span',class_="secondaryInfo")
years                    

[<span class="secondaryInfo">(1987)</span>,
 <span class="secondaryInfo">(2003)</span>,
 <span class="secondaryInfo">(2018)</span>,
 <span class="secondaryInfo">(1979)</span>,
 <span class="secondaryInfo">(2018)</span>,
 <span class="secondaryInfo">(1959)</span>,
 <span class="secondaryInfo">(1993)</span>,
 <span class="secondaryInfo">(1989)</span>,
 <span class="secondaryInfo">(2016)</span>,
 <span class="secondaryInfo">(1955)</span>,
 <span class="secondaryInfo">(2018)</span>,
 <span class="secondaryInfo">(1992)</span>,
 <span class="secondaryInfo">(2004)</span>,
 <span class="secondaryInfo">(2019)</span>,
 <span class="secondaryInfo">(2015)</span>,
 <span class="secondaryInfo">(2009)</span>,
 <span class="secondaryInfo">(2020)</span>,
 <span class="secondaryInfo">(2019)</span>,
 <span class="secondaryInfo">(2007)</span>,
 <span class="secondaryInfo">(2021)</span>,
 <span class="secondaryInfo">(2019)</span>,
 <span class="secondaryInfo">(1991)</span>,
 <span class="secondaryInfo">(20

In [12]:
Series_year=[]
for i in years:
    Series_year.append(i.text.replace("\n",""))
Series_year


['(1987)',
 '(2003)',
 '(2018)',
 '(1979)',
 '(2018)',
 '(1959)',
 '(1993)',
 '(1989)',
 '(2016)',
 '(1955)',
 '(2018)',
 '(1992)',
 '(2004)',
 '(2019)',
 '(2015)',
 '(2009)',
 '(2020)',
 '(2019)',
 '(2007)',
 '(2021)',
 '(2019)',
 '(1991)',
 '(2019)',
 '(2016)',
 '(2021)',
 '(1993)',
 '(2018)',
 '(1956)',
 '(1983)',
 '(1957)',
 '(2018)',
 '(2018)',
 '(1965)',
 '(2015)',
 '(2002)',
 '(1975)',
 '(1997)',
 '(1995)',
 '(2019)',
 '(2019)',
 '(2018)',
 '(2016)',
 '(2013)',
 '(2006)',
 '(2021)',
 '(2006)',
 '(2017)',
 '(2018)',
 '(2015)',
 '(2005)',
 '(2014)',
 '(1999)',
 '(2014)',
 '(1971)',
 '(1998)',
 '(2018)',
 '(2015)',
 '(2016)',
 '(2013)',
 '(2012)',
 '(2014)',
 '(2003)',
 '(2012)',
 '(2012)',
 '(2013)',
 '(2000)',
 '(2016)',
 '(1975)',
 '(2004)',
 '(2015)',
 '(2007)',
 '(2021)',
 '(2016)',
 '(2012)',
 '(2005)',
 '(2011)',
 '(1960)',
 '(1992)',
 '(2015)',
 '(2015)',
 '(2017)',
 '(2010)',
 '(2019)',
 '(1964)',
 '(2008)',
 '(2019)',
 '(2015)',
 '(1995)',
 '(2003)',
 '(2000)',
 '(2013)',

In [13]:
ranks=[]
for i in range(1,len(Series_name)+1):
    ranks.append(i)
ranks    

[1,
 2,
 3,
 4,
 5,
 6,
 7,
 8,
 9,
 10,
 11,
 12,
 13,
 14,
 15,
 16,
 17,
 18,
 19,
 20,
 21,
 22,
 23,
 24,
 25,
 26,
 27,
 28,
 29,
 30,
 31,
 32,
 33,
 34,
 35,
 36,
 37,
 38,
 39,
 40,
 41,
 42,
 43,
 44,
 45,
 46,
 47,
 48,
 49,
 50,
 51,
 52,
 53,
 54,
 55,
 56,
 57,
 58,
 59,
 60,
 61,
 62,
 63,
 64,
 65,
 66,
 67,
 68,
 69,
 70,
 71,
 72,
 73,
 74,
 75,
 76,
 77,
 78,
 79,
 80,
 81,
 82,
 83,
 84,
 85,
 86,
 87,
 88,
 89,
 90,
 91,
 92,
 93,
 94,
 95,
 96,
 97,
 98,
 99,
 100,
 101,
 102,
 103,
 104,
 105,
 106,
 107,
 108,
 109,
 110,
 111,
 112,
 113,
 114,
 115,
 116,
 117,
 118,
 119,
 120,
 121,
 122,
 123,
 124,
 125,
 126,
 127,
 128,
 129,
 130,
 131,
 132,
 133,
 134,
 135,
 136,
 137,
 138,
 139,
 140,
 141,
 142,
 143,
 144,
 145,
 146,
 147,
 148,
 149,
 150,
 151,
 152,
 153,
 154,
 155,
 156,
 157,
 158,
 159,
 160,
 161,
 162,
 163,
 164,
 165,
 166,
 167,
 168,
 169,
 170,
 171,
 172,
 173,
 174,
 175,
 176,
 177,
 178,
 179,
 180,
 181,
 182,
 183,
 184,
 185

In [14]:
import pandas as pd
topmovies=pd.DataFrame({})
topmovies['MOVIE rank']=ranks
topmovies['Top Indian Movie Names']=Series_name
topmovies[' IMDB rating']=Series_rating
topmovies['year of release']=Series_year

topmovies.style.hide_index()

In [15]:
len(topmovies)

250

There are 250 indian top rated movies available in the website

In [16]:
import pandas as pd
topmovies=pd.DataFrame({})
topmovies['MOVIE rank']=ranks[:100]
topmovies['Top Indian Movie Names']=Series_name[:100]
topmovies[' IMDB rating']=Series_rating[:100]
topmovies['year of release']=Series_year[:100]

topmovies.style.hide_index()

These are  100 Top Rated Indian Movies